modules/graceful/manager_unix.go

```diff
@@ -31,19 +32,19 @@ type gracefulManager struct {
 	terminateWaitGroup     sync.WaitGroup
 }
 
-func newGracefulManager() *gracefulManager {
+func newGracefulManager(ctx context.Context) *gracefulManager {
 	manager := &gracefulManager{
 		isChild: len(os.Getenv(listenFDs)) > 0 && os.Getppid() > 1,
```

Comment:
So, with a graceful restart there's a short period of time when `os.Getppid()` will return > 1 even if it's a new instance of Gitea (i.e. not a child): that is before the launching process (previous instance) had managed to exit; until then `os.Getppid()` will return the PID of the previous instance. Since the `gracefulManager` is one of the first things that are created when Gitea starts, is there a chance for `isChild` to set to `true` even when it should be `false`?

Reply:
That doesn't break this test - in fact it relies on ppid >0. 

We need ppid >0 to prevent Gitea trying to kill init(). In particular when Gitea is started by systemd it will pass in sockets with LISTEN_FDS set exactly like this - the only thing is its ppid will be 0.

The purpose of isChild is simply to say that we should attempt to kill the parent in future if it hasn't already died when we are ready.

Reply:
> The purpose of isChild is simply to say that we should attempt to kill the parent in future if it hasn't already died when we are ready.

Oh, OK. Yes, I remember we've talked about this, but I couldn't find our conversation with a quick search in the other PR.